go-runtime/ftl/testdata/go/typeregistry/typeregistry_test.go

```diff
@@ -0,0 +1,68 @@
+package typeregistry
+
+import (
+	"ftl/builtin"
+	"testing"
+
+	"github.com/TBD54566975/ftl/go-runtime/encoding"
+	"github.com/TBD54566975/ftl/go-runtime/ftl"
+	"github.com/TBD54566975/ftl/go-runtime/ftl/ftltest"
+	"github.com/alecthomas/assert"
+)
+
+var testCases = []struct {
+	Name    string
+	Input   StringsTypeEnum
+	Encoded string
+}{
+	{
+		Name:    "List",
+		Input:   List([]string{"asdf", "qwerty"}),
+		Encoded: `{"input":{"name":"List","value":["asdf","qwerty"]}}`,
+	},
+	{
+		Name:    "Single",
+		Input:   Single("asdf"),
+		Encoded: `{"input":{"name":"Single","value":"asdf"}}`,
+	},
+	{
+		Name:    "Object",
+		Input:   Object{S: "asdf"},
+		Encoded: `{"input":{"name":"Object","value":{"s":"asdf"}}}`,
+	},
+}
+
+func TestIngress(t *testing.T) {
+	ctx := ftltest.Context(ftltest.WithCallsAllowedWithinModule())
+
+	for _, test := range testCases {
```

Comment:
Pull the list of structs into the test function so it's self-contained.

Reply:
done